src/blockdata/script.rs

```diff
@@ -51,9 +53,79 @@ use crate::schnorr::{TapTweak, TweakedPublicKey, UntweakedPublicKey};
 /// ### Bitcoin Core References
 ///
 /// * [CScript definition](https://github.com/bitcoin/bitcoin/blob/d492dc1cdaabdc52b0766bf4cba4bd73178325d0/src/script/script.h#L410)
-#[derive(Clone, Default, PartialOrd, Ord, PartialEq, Eq, Hash)]
-pub struct Script(Box<[u8]>);
+#[derive(PartialOrd, Ord, PartialEq, Eq, Hash)]
+pub struct Script([u8]);
```

Comment:
Add `#[repr(transparent)]` to allow the casts?

Reply:
I don't believe it helps. `#[repr(transparent)]` only affects the relationship between newtypes and their encapsulated types, not references to them. However this looks promising: https://github.com/rust-lang/reference/pull/1206

Reply:
Thanks you! Looks like [`std` has the same issue](https://github.com/rust-lang/rust/blob/792bc5a0102d0973d42183a2b267850bb905236f/library/std/src/path.rs#L1162-L1167) however being `std` it can afford to do what it does and us being external library should do the opposite: use `#[repr(transparent)]` and document it's an implementation detail, right?